addons/html_builder/static/src/sidebar/block_tab.js

```diff
@@ -134,4 +153,223 @@ export class BlockTab extends Component {
+                this.state.ongoingInsertion = true;
+
+                const category = element.closest(".o_snippets_container").id;
+                const id = element.dataset.id;
+                snippet = this.snippetModel.getSnippet(category, id);
+                snippetEl = snippet.content.cloneNode(true);
+                isSnippetGroup = category === "snippet_groups";
+
+                // Check if the snippet is inline. Add it temporarily to the
+                // page to compute its style and get its `display` property.
+                this.document.body.appendChild(snippetEl);
+                const snippetStyle = window.getComputedStyle(snippetEl);
+                const isInlineSnippet = snippetStyle.display.includes("inline");
+                snippetEl.remove();
+
+                // Color-customize the snippet dynamic SVGs with the current
+                // theme colors.
+                const dynamicSvgEls = [
+                    ...snippetEl.querySelectorAll(
+                        'img[src^="/html_editor/shape/"], img[src^="/web_editor/shape/"]'
+                    ),
+                ];
+                dynamicSvgEls.forEach((dynamicSvgEl) => {
+                    const colorCustomizedURL = new URL(
+                        dynamicSvgEl.getAttribute("src"),
+                        window.location.origin
+                    );
+                    colorCustomizedURL.searchParams.forEach((value, key) => {
+                        const match = key.match(/^c([1-5])$/);
+                        if (match) {
+                            colorCustomizedURL.searchParams.set(
+                                key,
+                                getCSSVariableValue(`o-color-${match[1]}`)
+                            );
+                        }
+                    });
+                    dynamicSvgEl.src = colorCustomizedURL.pathname + colorCustomizedURL.search;
+                });
+
+                const selectors = this.shared.dropzone.getSelectors(snippetEl, snippet);
+                dropzoneEls = this.shared.dropzone.activateDropzones(selectors, {
+                    toInsertInline: isInlineSnippet,
+                });
+                this.onDropzoneStart();
+            },
+            dropzoneOver: ({ dropzone }) => {
+                const dropzoneEl = dropzone.el;
+                if (isSnippetGroup) {
+                    dropzoneEl.classList.add("o_dropzone_highlighted");
+                    return;
+                }
+                dropzoneEl.after(snippetEl);
+                dropzoneEl.classList.add("invisible");
+                this.onDropZoneOver();
+
+                // Preview the snippet correctly.
+                // Note: no async previews, in order to not slow down the drag.
+                this.cancelSnippetPreview = this.shared.history.makeSavePoint();
+                this.resources["on_snippet_preview_handlers"]?.forEach((onSnippetPreview) =>
```

Comment:
maybe, it would be better to call a shared function from dropzoneplugin instead. and that's a more general statement. the dropzone plugin should be the main coordinator of the drag and drop process, not the block tab.